pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -153,11 +164,17 @@ protected BasePartitionUpsertMetadataManager(String tableNameWithType, int parti
     _metadataTTL = context.getMetadataTTL();
     _deletedKeysTTL = context.getDeletedKeysTTL();
     _tableIndexDir = context.getTableIndexDir();
+    long trackingTimeMs = context.getNewSegmentTrackingTimeMs();
     UpsertConfig.ConsistencyMode cmode = context.getConsistencyMode();
     if (cmode == UpsertConfig.ConsistencyMode.SYNC || cmode == UpsertConfig.ConsistencyMode.SNAPSHOT) {
       _upsertViewManager = new UpsertViewManager(cmode, context);
+      // For consistency mode, we have to track newly added segments, so use default tracking time to enable the
+      // tracking of newly added segments if it's not enabled explicitly.
+      _newSegmentTrackingTimeMs =
+          trackingTimeMs > 0 ? trackingTimeMs : UpsertViewManager.DEFAULT_NEW_SEGMENT_TRACKING_TIME_MS;
```

Comment:
Consider banning this config in table validation, instead of ignoring the config that explicitly turns off this feature

Reply:
sgtm, just that the new validation check can break current users of consistency mode, so need to call it out.